pkg/db/db.go

```diff
@@ -0,0 +1,413 @@
+		| test  | 0          | PRIMARY  | 1            | id          | A         | 0           | NULL     | NULL   |      | BTREE      |         |               |
+		| test  | 0          | aid      | 1            | aid         | A         | 0           | NULL     | NULL   | YES  | BTREE      |         |               |
+		+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
+	*/
+
+	query := fmt.Sprintf("SHOW INDEX FROM `%s`.`%s`", dbname, table)
+	rows, err := db.Query(query)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer rows.Close()
+
+	var rowsData []map[string][]byte
+	for rows.Next() {
+		fields, err1 := ScanRow(rows)
+		if err1 != nil {
+			return nil, errors.Trace(err1)
+		}
+		rowsData = append(rowsData, fields)
+	}
+
+	return rowsData, nil
+}
+
+func FindSuitableIndex(db *sql.DB, dbName string, table string) (*model.ColumnInfo, error) {
+	rowsData, err := ShowIndex(db, dbName, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	tableInfo, err := GetSchemaTable(db, dbName, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// seek pk
+	for _, fields := range rowsData {
+		if string(fields["Key_name"]) == "PRIMARY" && string(fields["Seq_in_index"]) == "1" {
+			column, valid := GetColumnByName(tableInfo, string(fields["Column_name"]))
+			if !valid {
+				return nil, errors.New(fmt.Sprintf("can't find column %s in %s.%s", string(fields["Column_name"]), dbName, table))
+			}
+			return column, nil
+		}
+	}
+
+	// no pk found, seek unique index
+	for _, fields := range rowsData {
+		if string(fields["Non_unique"]) == "0" && string(fields["Seq_in_index"]) == "1" {
+			column, valid := GetColumnByName(tableInfo, string(fields["Column_name"]))
+			if !valid {
+				return nil, errors.New(fmt.Sprintf("can't find column %s in %s.%s", string(fields["Column_name"]), dbName, table))
+			}
+			return column, nil
+		}
+	}
+
+	// no unique index found, seek index with max cardinality
+	var c *model.ColumnInfo
+	var maxCardinality int
+	for _, fields := range rowsData {
+		if string(fields["Seq_in_index"]) == "1" {
+			cardinality, err := strconv.Atoi(string(fields["Cardinality"]))
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if cardinality > maxCardinality {
+				column, valid := GetColumnByName(tableInfo, string(fields["Column_name"]))
+				if !valid {
+					return nil, errors.New(fmt.Sprintf("can't find column %s in %s.%s", string(fields["Column_name"]), dbName, table))
+				}
+				maxCardinality = cardinality
+				c = column
+			}
+		}
+	}
+
+	return c, nil
+}
+
+// GetOrderKey return some columns for order
+func GetOrderKey(tbInfo *model.TableInfo) ([]string, []*model.ColumnInfo) {
+	keys := make([]string, 0, 2)
+	keyCols := make([]*model.ColumnInfo, 0, 2)
+
+	for _, index := range tbInfo.Indices {
+		if index.Primary {
+			for _, indexCol := range index.Columns {
```

Comment:
use `indexCol` to find column?

Reply:
are order keys and fit index  same?

Reply:
no, fit index just used to split data, and only use one column.
order by key used to sort data, and there will be more than one order by key.

Reply:
right

Reply:
if we have Implicit key, choose it preferentially